core/trino-main/src/main/java/io/trino/operator/join/BigintPagesHash.java

```diff
@@ -55,6 +57,17 @@
     private final long[] values;
     private final long size;
 
+    /**
+     * An estimated additional amount of memory (in bytes) required to create a page hash over a certain data set
+     */
+    public static long getEstimatedAdditionalSizeInBytes(
```

Comment:
Could we make it compute total (not only additional) by passing PagesIndex#getEstimatedSize() as param too?
It would feel more complete.

Same comment as in https://github.com/trinodb/trino/pull/15672/commits/31b33c10ca885720d3e9977d8c9f5ba55e1285f8#r1072870454